### PR TITLE
[SYCL][L0] Optimize sync of an in-order queue

### PR DESCRIPTION
**Pull request by smaslov-intel**

### CODE REVIEW

**Comment by smaslov-intel** [comment]
@jandres742 : please review

File: sycl/plugins/level_zero/pi_level_zero.cpp [hunk 1]
```diff
@@ -5726,17 +5726,28 @@ pi_result _pi_queue::synchronize() {
...
-        for (auto &ZeQueue : QueueGroup.second.ZeQueues)
-          if (ZeQueue)
-            ZE_CALL(zeHostSynchronize, (ZeQueue));
+  // Do nothing if the queue is empty
+  if (!LastCommandEvent)
+    return PI_SUCCESS;
+
+  // For in-order queue just wait for the last command.
+  if (isInOrderQueue()) {
+    ZE_CALL(zeHostSynchronize, (LastCommandEvent->ZeEvent));
```
Comment (jandres742) [inline sycl/plugins/level_zero/pi_level_zero.cpp:5735]: this works only if LastCommandEvent is the true last event of all the objects in QueueMap::QueueGroup. if so, then this is good.

Comment (smaslov-intel) [reply sycl/plugins/level_zero/pi_level_zero.cpp:5735]: yes, that's what is in `LastCommandEvent`, the last command submitted to the PI/UR queue.

**Review by againull — APPROVED** [review]

**Comment by smaslov-intel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1652

**Pull request merged**